<launch>
    <arg name="stdout" default="screen"/>
    <arg name="useDemo" default="false"/>

    <arg name="camera_name" default="vmpr"/>

    <arg name="is_multi_exp" default="false"/>
    <arg name="is_auto_mode" default="false"/>
	<arg name="camera_info_topic" default="/$(arg camera_name)_camera_color/camera_info"/>
    <arg name="ns_l" default="depth_camera"/>
    <arg name="ns_c" default="rgb"/>

    <arg name="image_frame_id" default="$(arg camera_name)_camera_color_optical_frame"/>
    <arg name="cloud_frame_id" default="$(arg camera_name)_camera_depth_optical_frame"/>
    <!-- <arg name="cloud_frame_id" default="sick_visionary_t_mini_depth_optical_frame"/> -->

    <arg name="image_mount_link_frame_id" default="$(arg camera_name)_camera_triton_mount_link"/>
    <arg name="cloud_mount_link_frame_id" default="$(arg camera_name)_camera_helios2_mount_link"/>
    <!-- <arg name="cloud_mount_link_frame_id" default="sick_visionary_t_mini_depth_mount_link"/> -->


    <arg name="featureFileName" default="features_info_$(arg ns_l)_to_$(arg ns_c)"/>
    <arg name="calib_result_name" default="$(arg ns_l)_to_$(arg ns_c)"/>

    <node pkg="lvt2calib" type="extrinsic_calib_l2c" name="extrinsic_calib_l2c" output="screen">
        <param name="ns_l" value="$(arg ns_l)"/>
        <param name="ns_c" value="$(arg ns_c)"/>

        <param name="image_frame_id" value="$(arg image_frame_id)"/>
        <param name="cloud_frame_id" value="$(arg cloud_frame_id)"/>

        <param name="image_mount_link_frame_id" value="$(arg image_mount_link_frame_id)"/>
        <param name="cloud_mount_link_frame_id" value="$(arg cloud_mount_link_frame_id)"/>

        <param name="save_calib_file" value="true"/>
        <param name="is_multi_exp" value="$(arg is_multi_exp)"/>
        <param name="is_auto_mode" value="$(arg is_auto_mode)"/>
       
        <!-- for Demo -->
        <param name="calib_result_dir_" type="string" value="$(find lvt2calib)/data/demo/calibration_result/" if="$(arg useDemo)"/>
        <param name="camera_info_topic_" type="string" value="$(arg camera_info_topic)" if="$(arg useDemo)"/>
        <param name="features_info_dir_" type="string" value="$(find lvt2calib)/data/demo/pattern_collection_result/features_info_demo.csv" if="$(arg useDemo)"/>
        
        <!-- for applictaion -->
        <param name="calib_result_dir_" type="string" value="$(find lvt2calib)/data/calibration_result/" unless="$(arg useDemo)"/>
        <param name="camera_info_topic_" type="string" value="$(arg camera_info_topic)" unless="$(arg useDemo)"/>
        <param name="features_info_dir_" type="string" value="$(find lvt2calib)/data/pattern_collection_result/$(arg featureFileName).csv" unless="$(arg useDemo)"/> 
        <param name="calib_result_name_" type="string" value="$(arg calib_result_name)" unless="$(arg useDemo)"/>
       
    </node>
</launch> 